drivers/builtin/src/nist_kw.c

```diff
@@ -124,82 +55,116 @@ static void calc_a_xor_t(unsigned char A[KW_SEMIBLOCK_LENGTH], uint64_t t)
  * KW-AE as defined in SP 800-38F section 6.2
  * KWP-AE as defined in SP 800-38F section 6.3
  */
-int mbedtls_nist_kw_wrap(mbedtls_nist_kw_context *ctx,
-                         mbedtls_nist_kw_mode_t mode,
-                         const unsigned char *input, size_t in_len,
-                         unsigned char *output, size_t *out_len, size_t out_size)
+psa_status_t mbedtls_nist_kw_wrap(mbedtls_svc_key_id_t key,
+                                  mbedtls_nist_kw_mode_t mode,
+                                  const unsigned char *input, size_t input_length,
+                                  unsigned char *output, size_t output_size, size_t *output_length)
 {
-    int ret = 0;
-    size_t semiblocks = 0;
-    size_t s;
-    size_t olen, padlen = 0;
+    psa_status_t ret = 0;
+    size_t semiblocks = 0, s, olen, padlen = 0, update_output_length, finish_output_length,
+           part_length;
     uint64_t t = 0;
     unsigned char outbuff[KW_SEMIBLOCK_LENGTH * 2];
     unsigned char inbuff[KW_SEMIBLOCK_LENGTH * 2];
+    psa_cipher_operation_t wrap_operation = PSA_CIPHER_OPERATION_INIT;
+    psa_key_attributes_t attributes;
+
+    ret = psa_get_key_attributes(key, &attributes);
```

Comment:
The type checks in wrap and unwrap are the same. Consider making them an auxiliary function.